grakn/rpc/session.py

```diff
@@ -47,7 +50,10 @@ def __init__(self, client, database: str, session_type: SessionType, options=Gra
 
         self._session_id = self._grpc_stub.session_open(open_req).session_id
         self._is_open = True
-        self._pulse = self._scheduler.enter(5, 1, self._transmit_pulse, ())
+        self._scheduler.enter(5, 1, self._transmit_pulse, ())
+        thread = Thread(target=self._scheduler.run)
+        # TODO: We should probably kill this thread when the session closes as it prevents the process closing.
```

Comment:
@alexjpwalker I don't think we can _kill_ thread in Python - it can only stop itself (by ceasing to run), see [ref](https://stackoverflow.com/questions/323972/is-there-any-way-to-kill-a-thread)

Reply:
Ah I see. Would cancelling all the scheduled tasks cause the thread to close? I think it would - will update the comment accordingly